config/database.yml

```diff
@@ -21,9 +21,12 @@ default: &default
   # http://guides.rubyonrails.org/configuring.html#database-pooling
   pool: 5
   host: localhost
-  username: root
   database: commons
 
+production:
+  <<: *default
+  username: root
```

Comment:
To clarify, for my second q, I'm asking why the production details itself are necessary at all, not just the username.